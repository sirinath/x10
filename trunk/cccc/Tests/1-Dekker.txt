# Dekker example from Doug Lea's and Vijay Saraswat's writeups

1 1 w x 1
1 2 r y 0
2 3 w y 1
2 4 r x 0

1.2 -> 2.3
2.4 -> 1.1
