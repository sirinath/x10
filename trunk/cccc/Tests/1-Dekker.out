1 1 w x 1
1 2 r y 0
2 3 w y 1
2 4 r x 0
# Manually specified dependencies
1.2 -> 2.3
2.4 -> 1.1
# Program-order dependencies
1.1 -> 1.2
2.3 -> 2.4
# Propagated dependencies
1.1 -> 2.3
1.2 -> 2.4
2.3 -> 1.1
2.4 -> 1.2
Happens-before dependencies for task 1
1.1 -> 1.2
1.1 -> 2.3
1.2 -> 2.3
2.3 -> 1.1
Cycle located:
1.1 -> 1.2  (program order)
1.2 -> 2.3  (manual)
2.3 -> 1.1  (propagation)
Task 1 sees cycle
Happens-before dependencies for task 2
1.1 -> 2.3
2.3 -> 1.1
2.3 -> 2.4
2.4 -> 1.1
Cycle located:
1.1 -> 2.3  (propagation)
2.3 -> 1.1  (propagation)
Task 2 sees cycle
Cycle found: illegal CCCC execution
